### ISSUE TITLE
[v3] Implement new Concept Exercise: `Higher-order functions`

### ISSUE THREAD

**Issue by porkostomus** [exercism/clojure#333]
This issue describes how to implement the `Higher-order functions` concept exercise for the Clojure track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how Higher-order functions are implemented in Clojure. 

## Learning objectives

The student will gain a general understanding of:

- First Class Functions
- Functions that take one or more functions as arguments
- Functions that return a function as its result

## Out of scope

- This is a general overview of the concept, not in-depth usage of specific functions.

## Concepts

- `higher-order-functions`

## Prerequisites

- `basics`

## Resources to refer to

[Clojure guide - Higher Order Functions](https://clojure.org/guides/higher_order_functions)

## Representer

The [Clojure Representer](https://github.com/exercism/clojure-representer) is currently being implemented and contributions are welcome.

## Analyzer

The [Clojure Analyzer](https://github.com/exercism/clojure-analyzer) is currently being implemented and contributions are welcome.

## Implementing

To implement a concept exercise, the following files must be created:

```
languages
└── clojure
    └── exercises
        └── concept
            └── <SLUG>
                ├── .docs
                |   ├── instructions.md
                |   ├── introduction.md
                |   ├── hints.md
                |   └── after.md (optional)
                ├── .meta
                |   |── design.md
                |   └── Example.clj
                ├── <NAME>.clj
                └── <NAME>_test.clj
```

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
